src/System.Management/src/System/Management/ManagementDateTime.cs

```diff
@@ -195,16 +195,13 @@ public static DateTime ToDateTime(string dmtfDate)
             }
 
 
-            // Construct a new System.DateTime object
-            var datetime = new DateTime(year, month, day, hour, minute, second, 0, DateTimeKind.Utc);
+            // Construct a new System.DateTime object, netfx uses date kind unspecified so use the same
+            var datetime = new DateTime(year, month, day, hour, minute, second, 0, DateTimeKind.Unspecified);
             // Then add the ticks calculated from the microseconds
             datetime = datetime.AddTicks(ticks);
-            // Then subtruct offset in minutes
-            datetime = datetime.AddMinutes(-utcOffset);
+            // Then adjust the offset, using a manual calulation to keep the same possible range as netfx
+            datetime = datetime.AddMinutes(-(utcOffset - TimeZoneInfo.Local.GetUtcOffset(datetime).Ticks / TimeSpan.TicksPerMinute));
```

Comment:
@tarekgh in the end I had to go back to the manual calculation since in the typical case of same time zone this can lead to out-of-range for values near DateTime.MinValue (granted that this is a corner case, but anyway then we succeed in the same cases as netfx).

Reply:
I am wondering why this happened. your calculations should produce the same result as ToLocalTime(). do you have some example of such case?

Reply:
The issue is with values near DateTime.MinValue, if we AddMinutes(-utcOffset) from a time zone ahead of UTC then we can get an out-of-range error before we can call ToLocalTime to make the proper adjustment. The test added is an example of that, without the manual calculation, it will fail depending on the time zone.